src/com/trilead/ssh2/Connection.java

```diff
@@ -1491,7 +1491,7 @@ public synchronized void enableDebugging(boolean enable, DebugLogger logger)
 		{
 			if (logger == null)
 			{
-				logger = new DebugLogger()
+				Logger.logger = new DebugLogger()
```

Comment:
This doesn't seem correct: it only sets a logger if a null was passed for the `logger` parameter.

I think this should be a change to do `Logger.logger = logger;` as the last statement in this else block (just outside the `if (logger == null)` check) and leave the setting of this variable as it was, or add an `else` to the `if (logger == null)` block containing `Logger.logger = logger;`